### PR TITLE
Feat: add sorting/filtering to table

### PR DESCRIPTION
**Pull request by mykalmax**

### CODE REVIEW

File: web/client/src/library/components/table/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+                style={{ height: `${MIN_HEIGHT_ROW}px` }}
+              >
+                {headerGroup.headers.map(header => (
+                  <th
+                    key={header.id}
+                    className={clsx(
+                      'pl-2 pr-4 pt-2 text-sm pb-1 border-r-2 last:border-r-0 border-light dark:border-dark',
+                    )}
+                  >
+                    {header.isPlaceholder ? null : (
```
Comment (vchan) [inline web/client/src/library/components/table/Table.tsx:114]: If it's a placeholder, does the word `null` appear on the screen?

File: web/client/src/library/components/table/help.ts [hunk 2]
```diff
@@ -1,185 +1,44 @@
...
-function isModified(diff: any, header: string, key: string): boolean {
-  const source_sample = diff.row_diff.sample[`${SOURCE_PREFIX}${header}`]?.[key]
-  const target_sample = diff.row_diff.sample[`${TARGET_PREFIX}${header}`]?.[key]
-
-  return (
-    (isNotNil(source_sample) || isNotNil(target_sample)) &&
-    source_sample !== target_sample
-  )
-}
+  console.log('getTableDataFromArrowStreamResult', result)
```
Comment (vchan) [inline web/client/src/library/components/table/help.ts:17]:
```suggestion
```

File: web/client/src/library/components/tableDiff/help.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  return (
+    diff.row_diff.sample[`${SOURCE_PREFIX}${header}`]?.[key] ?? EMTPY_TABLE_CELL
+  )
+}
+
+function getCellContentTarget(diff: any, header: string, key: string): string {
+  return (
+    diff.row_diff.sample[`${TARGET_PREFIX}${header}`]?.[key] ?? EMTPY_TABLE_CELL
+  )
+}
```
Comment (vchan) [inline web/client/src/library/components/tableDiff/help.ts:185]: How do you feel about refactoring these to maybe
```Typescript
function getCellContent(diff: any, prefix: string, header: string, key: string): string {
  return (
    diff.row_diff.sample[`${prefix}${header}`]?.[key]
  )
}

function getCellContentSource(diff: any, header: string, key: string): string {
  return getCellContent(diff, SOURCE_PREFIX, header, key) ?? EMPTY_TABLE_CELL
}

function getCellContentTarget(diff: any, header: string, key: string): string {
  return getCellContent(diff, TARGET_PREFIX, header, key) ?? EMPTY_TABLE_CELL
}
```
and also using them in the other functions in this file?

File: web/client/src/library/components/tableDiff/help.ts [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+  }
+}
+
+function isModified(diff: any, header: string, key: string): boolean {
+  const source_sample = diff.row_diff.sample[`${SOURCE_PREFIX}${header}`]?.[key]
+  const target_sample = diff.row_diff.sample[`${TARGET_PREFIX}${header}`]?.[key]
+
+  return (
+    (isNotNil(source_sample) || isNotNil(target_sample)) &&
+    source_sample !== target_sample
```
Comment (vchan) [inline web/client/src/library/components/tableDiff/help.ts:106]: Is just `source_sample !== target_sample` sufficient?

**Review by vchan — APPROVED** [review]

File: web/client/src/library/components/table/Table.tsx (hunk 1 above)
Comment (mykalmax) [reply web/client/src/library/components/table/Table.tsx:114]: no

**Pull request merged**